modules/Components/src/client/ACASFormFields.coffee

```diff
@@ -183,24 +183,73 @@ class window.ACASFormLSLabelFieldController extends ACASFormAbstractFieldControl
 		@userInputEvent = true
 		value = UtilityFunctions::getTrimmedInput(@$('input'))
 
-		if @isValid(value)
-			if value != ""
-				@getModel().set
-					labelText: value
-					ignored: false
-			else
-				@setEmptyValue()
-		super()
+		# check
+		@isValid value, =>
+			super()
 
-	isValid: (value) ->
+	isValid: (value, callback) =>
+		@checkUnique value, (isValid) =>
+			if isValid
+				@clearError()
+				if value != ""
+					@getModel().set
+						labelText: value
+						ignored: false
+				else
+					@setEmptyValue()
+			callback()
+
+	checkUnique: (value, callback) ->
+		if @getModel().get("validationRegex")?
+			regex = new RegExp(@getModel().get("validationRegex"));
+			if !regex.test(value)
+				@setError("label does not meet format requirements")
+				return
 		if value.length > @maxLabelLength
-			@setError("label is too long")
-			@setEmptyValue()
-			return false
+				@setError("label is too long")
+				@setEmptyValue()
+				callback false
+				return
+		
+		if @getModel().get("unique")? && @getModel().get("unique")
+			# if the value is empty, don't check for uniqueness, the form field should evaluate if this is required
+			# so we don't need to do that here
+			if(value == "") 
+				callback(true)
+				return
+			# check for uniqueness
+			$.ajax
+				type: 'POST'
+				url: "/api/getThingCodeByLabel/#{@getModel().get("uniqueBy").thingType}/#{@getModel().get("uniqueBy").thingKind}"
+				data: 
+					thingType: @getModel().get("uniqueBy").thingType
```

Comment:
Is there a reason this is passed in via "uniqueBy" rather than just directly setting the thingType and thingKind on the Label model? If we're setting new properties anyway, I find "uniqueBy" kindof misleading. I feel like I should need to pass in "labelText" or some other parameter to tell ACAS which field(s) to "unique by"